cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/template/etl/batch/ETLESTest.java

```diff
@@ -0,0 +1,195 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.template.etl.batch;
+
+import co.cask.cdap.api.data.format.Formats;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.dataset.table.Scanner;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.common.utils.Networks;
+import co.cask.cdap.proto.AdapterConfig;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.template.etl.batch.config.ETLBatchConfig;
+import co.cask.cdap.template.etl.batch.sink.ElasticsearchSink;
+import co.cask.cdap.template.etl.batch.source.ElasticsearchSource;
+import co.cask.cdap.template.etl.common.ETLStage;
+import co.cask.cdap.template.etl.common.Properties;
+import co.cask.cdap.test.AdapterManager;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.cdap.test.SlowTests;
+import co.cask.cdap.test.StreamManager;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest;
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexResponse;
+import org.elasticsearch.action.delete.DeleteResponse;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.settings.ImmutableSettings;
+import org.elasticsearch.node.Node;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
+
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+import static org.elasticsearch.node.NodeBuilder.nodeBuilder;
+
+/**
+ * <p>
+ *  Unit test for batch {@link ElasticsearchSink} and {@link ElasticsearchSource} classes.
+ * </p>
+ */
+public class ETLESTest extends BaseETLBatchTest {
+  private Client client;
+  private Node node;
+
+  private static final Gson GSON = new Gson();
+
+  private static final Schema BODY_SCHEMA = Schema.recordOf(
+    "event",
+    Schema.Field.of("ticker", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of("num", Schema.of(Schema.Type.INT)),
+    Schema.Field.of("price", Schema.of(Schema.Type.DOUBLE)));
+
+  @ClassRule
+  public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private int port;
+
+  @Before
+  public void beforeTest() throws Exception {
+    port = Networks.getRandomPort();
```

Comment:
It's not - it's the default. Elasticsearch defaults to 9200, but you can change that setting (which I do when creating the elasticsearch instance)